File: Item.cpp
#include "Item.h"

Item::Item(const std::string& name, int64_t price)
    : name(name), price(price), icon_png("") {}

Item::Item(const std::string& name, int64_t price, const std::string& icon_png)
    : name(name), price(price), icon_png(icon_png) {}

std::string Item::getName() const {
    return name;
}

int64_t Item::getPrice() const {
    return price;
}

std::string Item::getIconPng() const {
    return icon_png;
}

void Item::setName(const std::string& name) {
    this->name = name;
}

void Item::setPrice(int64_t price) {
    this->price = price;
}

void Item::setIconPng(const std::string& icon_png) {
    this->icon_png = icon_png;
}

File: src/item.rs
#[derive(Debug)]
pub struct Item {
    pub name: String,
    pub price: i64,
    pub icon_png: Option<String>,
}

impl Item {
    pub fn new(name: String, price: i64) -> Self {
        Self {
            name,
            price,
            icon_png: None,
        }
    }

    pub fn with_icon(name: String, price: i64, icon_png: String) -> Self {
        Self {
            name,
            price,
            icon_png: Some(icon_png),
        }
    }
}
